domain/model/service/modelservice.go

```diff
@@ -43,26 +56,68 @@ type ControllerState interface {
 	GetModelState(context.Context, coremodel.UUID) (model.ModelState, error)
 }
 
+// EnvironVersionProvider defines a minimal subset of the EnvironProvider interface
```

Comment:
Can you please move all the environ provider code - the stuff below, the ModelEnvironProvider interface, the implementation on the service - to a separate environprovider.go file. What we have here is effectively a placeholder / hack because we need the provider version and don't yet have a clear idea how we might want to model it in 4.0. Let's keep this code separate from the rest of the service implementation.

Add an extra comment at the top of the file stating that we need to environ version for export so that we can do the right thing in the future but it's not actively used at the moment so it's been decided to get it directly from the environ provider instead of the underlying data model.

Reply:
Sure :)

Reply:
hi ian, im not too sure how we could user test this change, would you mind sharing how i could go about this?